Base/QTCore/qSlicerCoreApplication.cxx

```diff
@@ -2396,3 +2396,9 @@ void qSlicerCoreApplication::openUrl(const QString& url)
 {
   emit urlReceived(url);
 }
+
+//------------------------------------------------------------------------------
+void qSlicerCoreApplication::logUsageEvent(const QString& component, const QString& event)
+{
+  emit usageEventLogged(component, event);
```

Comment:
Conditionally call this based on a build option, will be useful for custom application

Reply:
Consider skipping when testing is enabled ? and or if build is not in "Release" ?

Reply:
Added `Slicer_BUILD_USAGE_LOGGING_SUPPORT` CMake option to completely disable this feature (`logUsageEvent` has no effect if this flag is disabled).

Further logic would be more suitable to implement at lower levels:
- testing mode: If the feature was completely disabled in testing mode then it would be hard to test the feature. Testing mode may also be used for batch processing, where we may want get usage statistics from. Instead the modules that get the event (such as the data collector in Telemetry extension) could check the state of the application and process the events accordingly (e.g., number of events occurring during automatic testing could be collected independently from events that occur outside of testing mode).
- Debug/release mode: The C++ build mode does not necessarily reflect if the application is being developed. Release mode may be used for development, especially if modules are developed in Python. Also, "Developer mode" could be used for this purpose, too (but that may be enabled by too many people for too long time). Overall, this logic is not simple and would be too rigid to specify in the Slicer core application.